unittests/test_pipeline.py

```diff
@@ -170,6 +171,21 @@ def test_hellocheck_local(hellotest, local_exec_ctx):
         assert os.path.exists(os.path.join(hellotest.outputdir, f))
 
 
+def test_hellocheck_remote_building(testsys_system):
+    @fixtures.custom_prefix('unittests/resources/checks')
+    class MyTest(HelloTest):
+        def __init__(self):
+            super().__init__()
+            self.build_locally = False
+
+    test = MyTest()
+    partition = fixtures.partition_by_name('gpu')
+    environ = partition.environment('builtin-gcc')
+    test.setup(partition, environ)
+
+    assert isinstance(test._build_job, _SlurmJob)
```

Comment:
I think it's better if you run the test:

```suggestion
def test_hellocheck_remote_building(hellotest, remote_exec_ctx):
    hellotest.build_locally = False
    _run(hellotest, *remote_exec_ctx)    
```